Detectors/EMCAL/simulation/src/Detector.cxx

```diff
@@ -161,9 +162,9 @@ Hit* Detector::AddHit(Int_t trackID, Int_t parentID, Int_t primary, Double_t ini
               << ")  with energy " << initialEnergy << " loosing " << eLoss << std::endl;
 
   Hit* myhit(nullptr);
-  auto hitentry = mEventHits.find({ trackID, nullptr });
+  auto hitentry = mEventHits.find(trackID);
   if (hitentry != mEventHits.end()) {
-    myhit = hitentry->mHit;
+    myhit = hitentry->second;
```

Comment:
@mfasDa : I believe this mechanism might actually not be necessary. The Geant engine will ( *I will confirm this* ) finish processing a track until it moves to another track. So the trackID will essentially be the same until we change it to the next track. You can hence easily detect when a new track is seen. When this is the case you stop accumulation of the hit and start a new one. We use this mechanism in TPC: See `Detector.cxx:312.` In this case, it is not necessary to have this lookup structure.